### DISCUSSION TITLE
Notification channels

### DISCUSSION THREAD

**Discussion by Agio56** [NightscoutFoundation/xDrip#3628] [Q&A]
Hi @Navid200,
I still need your help...
If I enable "Notifications Channels" in "Alarm and Alerts", are all  settings made here overwritten by those made in Android settings (Settings->apps->xdrip->notifications)?
For example, in "Alarm and Alerts" I can set a different sound for hypoglycemia and hyperglycemia, while there is only one notification channel available for both.
I'm quite confused :(

**Comment by Navid200** [comment]
This is a very good question.   I hope I can give the right answer.

You can have multiple low glucose level alerts in xDrip and you can have multiple High glucose level alerts in xDrip.  Each one of them can have a unique sound file assigned to it be xDrip.  
However, xDrip does not assign a different notification channel to each.  All of them use the same "BG_ALERT_CHANNEL" notification channel.  
Not only that, xDrip does not even have a different notification channel for low glucose alerts versus High glucose level alerts.

As long as you don't assign a sound to the Glucose Alerts Android notification channel, your phone will respect the different sound file each one of your notifications plays.  

Now, let's look at something that is the exact opposite.
xDrip does not assign a unique sound file to missed reading alert.  At least, not yet.  xDrip missed reading alert notification uses the same sound file you can assign to fast rising and fast falling alerts.
But, xDrip assigns different notification channels to them!  xDrip uses the BG_RISE_DROP_CHANNEL for the fast rising or fast falling alerts.  But, xDrip uses the BG_MISSED_ALERT_CHANNEL for the missed reading alert.
If you assign different sound files to those notification channels in Android, your phone will play different sounds when they go off.  

In other words, if you use the notification channels, they can be very flexible.  

This may bring up the question of why would anyone disable notification channels in xDrip.
We don't have full control in xDrip on exactly what a phone may do with notification channels.
This is an example: https://github.com/NightscoutFoundation/xDrip/issues/1503  

If Notification channels work for you, you can benefit from enabling them.  

I give you another example advantage of notification channels.
You have no way of disabling vibration for fast rising, missed reading, forecast low or persistent high alerts in xDrip.
But, using notification channels, you can.

**Reply by Agio56** [reply]
Thank you, very helpful and very clear this time too.
